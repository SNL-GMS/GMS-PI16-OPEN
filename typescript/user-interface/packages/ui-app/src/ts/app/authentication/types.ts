import { UserSessionTypes } from '@gms/ui-state';

export interface Authenticator {
  authenticateWith(userName: string): Promise<UserSessionTypes.AuthStatus>;
  checkIsAuthenticated(): Promise<UserSessionTypes.AuthStatus>;
  unAuthenticateWith(): Promise<UserSessionTypes.AuthStatus>;
  logout(setAuthStatus: (status: UserSessionTypes.AuthStatus) => void): void;
}

export interface LoginRequestParams {
  userName: string;
}
