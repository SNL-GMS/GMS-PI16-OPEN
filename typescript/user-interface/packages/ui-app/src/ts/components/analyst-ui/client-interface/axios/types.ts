/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  ChannelSegmentTypes,
  EventTypes,
  QcMaskTypes,
  SignalDetectionTypes,
  WaveformTypes,
  WorkflowTypes
} from '@gms/common-model';
import { AxiosRequestConfig } from 'axios';
import { QueryResult } from 'react-query';

export interface EventMutationArgs {
  variables: {
    events: EventTypes.Event[];
  };
}

export interface QcMaskMutationArgs {
  variables: {
    qcMasks: QcMaskTypes.QcMask[];
  };
}

export interface SignalDetectionMutationArgs {
  variables: {
    signalDetections: SignalDetectionTypes.SignalDetection[];
  };
}

// ***************************************
// Workflow Interval Query and Mutation interfaces
// ***************************************
export interface StageIntervalsByStageIdAndTimeRequest {
  startTime: number;
  endTime: number;
  stageIds: WorkflowTypes.WorkflowDefinitionId[];
}
export interface UserRequest {
  userName: string;
  time: number;
}

export interface UpdateActivityIntervalStatusRequest extends UserRequest {
  activityIntervalId: WorkflowTypes.IntervalId;
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface UpdateInteractiveAnalysisStageIntervalStatusRequest extends UserRequest {
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface SplitActivityIntervalRequest {
  intervalId: WorkflowTypes.IntervalId;
  duration: number;
}
export interface EventQueryProps {
  eventQuery: QueryResult<EventTypes.Event[], any>;
}

export interface QcMaskQueryProps {
  qcMaskQuery: QueryResult<QcMaskTypes.QcMask[], any>;
}

export interface SignalDetectionQueryProps {
  signalDetectionQuery: QueryResult<SignalDetectionTypes.SignalDetection[], any>;
}

export interface WaveformQueryProps {
  waveformQuery: QueryResult<ChannelSegmentTypes.ChannelSegment<WaveformTypes.Waveform>[], any>;
}

/**
 * An axios request known to have a waveform query request in its data.
 */
export type WaveformAxiosRequestConfig = AxiosRequestConfig & {
  data: WaveformTypes.WaveformQueryArgs;
};
