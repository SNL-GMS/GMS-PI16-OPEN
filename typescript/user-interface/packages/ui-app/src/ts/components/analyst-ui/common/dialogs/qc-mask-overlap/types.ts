import { QcMaskTypes } from '@gms/common-model';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { CoordinatesPx } from '../qc-mask-form/types';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { QcMaskDialogBoxType } from '../types';

/**
 * QcMaskOverlap Props
 */
export interface QcMaskOverlapProps {
  masks: QcMaskTypes.QcMask[];
  // If set, any context menus opened as children will be created at given x/y coordinates
  contextMenuCoordinates?: CoordinatesPx;
  openNewContextMenu(
    x: number,
    y: number,
    mask: QcMaskTypes.QcMask,
    dialogType: QcMaskDialogBoxType
  );
  selectMask(mask: QcMaskTypes.QcMask);
}

/**
 * QcMaskOverlap  State
 */
export interface QcMaskOverlapState {
  selectedMask: QcMaskTypes.QcMask;
  createDialogOfType?: QcMaskDialogBoxType;
}
export interface QcMaskTableButtonParams {
  onClick(x: number, y: number, params: any);
}
