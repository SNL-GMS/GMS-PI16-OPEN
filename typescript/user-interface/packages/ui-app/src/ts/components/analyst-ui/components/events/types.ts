import { GQLTypes } from '@gms/common-graphql';
import { CommonTypes, EventTypes, WorkflowTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import { ChildProps, MutationFunction } from 'react-apollo';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction;
  saveEvent: MutationFunction;
  saveAllModifiedEvents: MutationFunction;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analysisMode: WorkflowTypes.AnalysisMode;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  GQLTypes.EventsInTimeRangeQueryProps &
  GQLTypes.DefaultStationsQueryProps &
  GQLTypes.SignalDetectionsByStationQueryProps &
  GQLTypes.WorkspaceStateProps;
