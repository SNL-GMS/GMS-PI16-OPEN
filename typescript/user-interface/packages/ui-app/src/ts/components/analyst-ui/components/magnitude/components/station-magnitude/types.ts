import { EventTypes } from '@gms/common-model';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import Immutable from 'immutable';
import { MutationFunction } from 'react-apollo';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { AmplitudesByStation } from '../../types';

/**
 * StationMagnitude State
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface StationMagnitudeState {
  computeNetworkMagnitudeSolutionStatus: Immutable.Map<
    string,
    [{ stationName: string; rational: string }]
  >;
}

export interface MagnitudeAndSdData {
  magSolution: EventTypes.NetworkMagnitudeSolution;
  sdData: StationMagnitudeSdData;
}

/**
 * options that can be passed to ag grid
 */
export interface Options {
  alignedGrids: any[];
}

/**
 * StationMagnitude Props
 */
export interface StationMagnitudeProps {
  options?: Options;
  amplitudesByStation: AmplitudesByStation[];
  historicalMode: boolean;
  selectedSdIds: string[];
  locationSolution: EventTypes.LocationSolution;
  displayedMagnitudeTypes: AnalystWorkspaceTypes.DisplayedMagnitudeTypes;
  computeNetworkMagnitudeSolution: MutationFunction;
  openEventId: string;

  setSelectedSdIds(ids: string[]): void;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<[{ stationName: string; rational: string }]>;
}

export interface StationMagnitudeSdData {
  channel: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  signalDetectionId: string;
  time: number;
  stationName: string;
  flagForReview: boolean;
}

export interface MagnitudeDataForRow {
  channel: string;
  signalDetectionId: string;
  phase: string;
  amplitudeValue: number;
  amplitudePeriod: number;
  flagForReview: boolean;
  defining: boolean;
  mag: number;
  res: number;
  hasMagnitudeCalculationError: boolean;
  computeNetworkMagnitudeSolutionStatus: string;
}

/**
 * Table row object for station magnitude
 */
export interface StationMagnitudeRow {
  id: string;
  dataForMagnitude: Map<EventTypes.MagnitudeType, MagnitudeDataForRow>;
  station: string;
  dist: number;
  azimuth: number;
  selectedSdIds: string[];
  historicalMode: boolean;
  azimuthTooltip: string;
  checkBoxCallback(
    magnitudeType: EventTypes.MagnitudeType,
    stationNames: string[],
    defining: boolean
  ): Promise<void>;
}
export interface StationMagAndSignalDetection {
  stationName: string;
  magnitudeAndSdData: Map<EventTypes.MagnitudeType, MagnitudeAndSdData>;
}
