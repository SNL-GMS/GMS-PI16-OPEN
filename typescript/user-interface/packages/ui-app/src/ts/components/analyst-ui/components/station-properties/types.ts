import { ChannelTypes, StationTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import {
  CellRendererParams,
  ColumnDefinition,
  Row,
  RowClickedEvent,
  ToolbarTypes
} from '@gms/ui-core-components';

export interface StationPropertiesComponentProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
}
export interface StationPropertiesPanelProps {
  selectedStation: string;
  effectiveAtTimes: string[];
}
export interface StationPropertiesToolbarProps {
  selectedEffectiveAt: string;
  selectedStation: StationTypes.Station;
  stationName: string;
  onEffectiveTimeChange: (args: any) => void;
  effectiveAtTimes: string[];
  widthPx?: number;
  setSelectedSiteColumnsToDisplay: (args: Map<siteColumnEnum, boolean>) => void;
  setSelectedChannelColumnsToDisplay: (args: Map<channelColumnEnum, boolean>) => void;
}

export interface SiteConfigurationTableProps {
  station: StationTypes.Station;
  onRowSelection(event: SiteConfigurationRowClickedEvent);
  columnsToDisplay: Map<string, boolean>;
}

export interface ChannelConfigurationTableProps {
  channels: ChannelTypes.Channel[];
  stationData: StationTypes.Station;
  columnsToDisplay: Map<string, boolean>;
}

export interface ChannelConfigurationRow extends Row {
  name: string;
  channelBandType;
  channelInstrumentType;
  channelOrientationType;
  channelOrientationCode;
  channelDataType;
  nominalSampleRateHz;
  description: string;
  calibrationFactor: string;
  calibrationStandardDeviation: string;
  calibrationPeriod: string;
  calibrationTimeShift: string;
  orientationHorizontalDegrees: string;
  orientationVerticalDegrees: string;
  latitudeDegrees: string;
  longitudeDegrees: string;
  depthKm: string;
  elevationKm: string;
  units: string;
  effectiveAt: string;
  effectiveUntil: string;
  calibrationEffectiveAt: string;
  fapResponseId: string;
  calibrationResponseId: string;
  northDisplacementKm: string;
  eastDisplacementKm: string;
  verticalDisplacementKm: string;
}

/**
 * Table row clicked event
 */
export type SiteConfigurationRowClickedEvent = RowClickedEvent<
  { id: string },
  any,
  number | string
>;

export interface SiteConfigurationRow extends Row {
  name: string;
  description: string;
  effectiveAt: string;
  effectiveUntil: string;
  latitudeDegrees: string;
  longitudeDegrees: string;
  elevationKm: string;
  depthKm: string;
  type: string;
}
export interface PropertiesToolbarItemDefs {
  rightToolbarItemDefs: ToolbarTypes.ToolbarItem[];
  leftToolbarItemDefs: ToolbarTypes.ToolbarItem[];
}

export type StationPropertiesColumnDefinition = ColumnDefinition<
  any, // RowDataType
  any, // ContextDataType
  string, // CellValueType
  any, // CellRendererParamsType
  any // HeaderRendererParamsType
>;

export type StationPropertiesCellRendererProps = CellRendererParams<
  SiteConfigurationRow | ChannelConfigurationRow, // RowDataType
  any, // ContextDataType
  string, // CellValueType
  any, // CellRendererParamsType
  {
    // HeaderRendererParamsType
    value: string | number;
    formattedValue: string;
  }
>;

/**
 * used to populate the values of the site column picker dropdown, and match the values to the table column ids
 */
export enum siteColumnEnum {
  name = 'name',
  effectiveAt = 'effectiveAt',
  effectiveUntil = 'effectiveUntil',
  latitudeDegrees = 'latitudeDegrees',
  longitudeDegrees = 'longitudeDegrees',
  depthKm = 'depthKm',
  elevationKm = 'elevationKm',
  description = 'description',
  type = 'type'
}

/**
 * used to match the display strings to values in the site table column picker dropdown
 */
export const siteColumnDisplayStrings: Map<string, string> = new Map<string, string>([
  ['name', 'Name'],
  ['effectiveAt', 'Effective At'],
  ['effectiveUntil', 'Effective Until'],
  ['latitudeDegrees', 'Latitude'],
  ['longitudeDegrees', 'Longitude'],
  ['depthKm', 'Depth'],
  ['elevationKm', 'Elevation'],
  ['description', 'Description'],
  ['type', 'Type']
]);

/**
 * used to match the display strings to values in the channel table column picker dropdown
 */
export const channelColumnDisplayStrings: Map<string, string> = new Map<string, string>([
  ['name', 'Name'],
  ['effectiveAt', 'Effective At'],
  ['effectiveUntil', 'Effective Until'],
  ['latitudeDegrees', 'Latitude'],
  ['longitudeDegrees', 'Longitude'],
  ['depthKm', 'Depth'],
  ['elevationKm', 'Elevation'],
  ['nominalSampleRateHz', 'Sample Rate'],
  ['units', 'Units'],
  ['orientationHorizontalDegrees', 'Horizontal Angle'],
  ['orientationVerticalDegrees', 'Vertical Angle'],
  ['calibrationFactor', 'Calibration Factor'],
  ['calibrationPeriod', 'Calibration Period'],
  ['calibrationEffectiveAt', 'Calibration Effective At'],
  ['calibrationTimeShift', 'Calibration Time Shift'],
  ['calibrationStandardDeviation', 'Calibration Std Dev'],
  ['northDisplacementKm', 'North Displacement'],
  ['eastDisplacementKm', 'East Displacement'],
  ['verticalDisplacementKm', 'Vertical Displacement'],
  ['description', 'Description'],
  ['channelDataType', 'Data Type'],
  ['channelBandType', 'Band Type'],
  ['channelInstrumentType', 'Instrument Type'],
  ['channelOrientationCode', 'Orientation Code'],
  ['channelOrientationType', 'Orientation Type'],
  ['calibrationResponseId', 'Calibration ID'],
  ['fapResponseId', 'FAP Resp ID']
]);

/**
 * used to populate the values of the channel column picker dropdown, and match the values to the table column ids
 */
export enum channelColumnEnum {
  name = 'name',
  effectiveAt = 'effectiveAt',
  effectiveUntil = 'effectiveUntil',
  latitudeDegrees = 'latitudeDegrees',
  longitudeDegrees = 'longitudeDegrees',
  depthKm = 'depthKm',
  elevationKm = 'elevationKm',
  nominalSampleRateHz = 'nominalSampleRateHz',
  units = 'units',
  orientationHorizontalDegrees = 'orientationHorizontalDegrees',
  orientationVerticalDegrees = 'orientationVerticalDegrees',
  calibrationFactor = 'calibrationFactor',
  calibrationPeriod = 'calibrationPeriod',
  calibrationEffectiveAt = 'calibrationEffectiveAt',
  calibrationTimeShift = 'calibrationTimeShift',
  calibrationStandardDeviation = 'calibrationStandardDeviation',
  northDisplacementKm = 'northDisplacementKm',
  eastDisplacementKm = 'eastDisplacementKm',
  verticalDisplacementKm = 'verticalDisplacementKm',
  description = 'description',
  channelDataType = 'channelDataType',
  channelBandType = 'channelBandType',
  channelInstrumentType = 'channelInstrumentType',
  channelOrientationCode = 'channelOrientationCode',
  channelOrientationType = 'channelOrientationType',
  calibrationResponseId = 'calibrationResponseId',
  fapResponseId = 'fapResponseId'
}
