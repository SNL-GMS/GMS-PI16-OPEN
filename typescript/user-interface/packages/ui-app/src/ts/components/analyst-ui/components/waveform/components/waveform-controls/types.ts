import { CommonTypes } from '@gms/common-model';
import { AnalystWorkspaceTypes } from '@gms/ui-state';
import React from 'react';

import { QcMaskDisplayFilters } from '~analyst-ui/config';
import { MaskDisplayFilter } from '~analyst-ui/config/user-preferences';

import { AlignWaveformsOn, PanType } from '../../types';
import { AmplitudeScalingOptions, FixedScaleValue } from './scaling-options';

/**
 * Waveform Display Controls Props
 */
export interface WaveformControlsProps {
  defaultSignalDetectionPhase: CommonTypes.PhaseType;
  currentSortType: AnalystWorkspaceTypes.WaveformSortType;
  currentOpenEventId: string;
  analystNumberOfWaveforms: number;
  showPredictedPhases: boolean;
  maskDisplayFilters: QcMaskDisplayFilters;
  alignWaveformsOn: AlignWaveformsOn;
  phaseToAlignOn: CommonTypes.PhaseType | undefined;
  alignablePhases: CommonTypes.PhaseType[] | undefined;
  isMeasureWindowVisible: boolean;
  measurementMode: AnalystWorkspaceTypes.MeasurementMode;
  setMode(mode: AnalystWorkspaceTypes.WaveformDisplayMode): void;
  setDefaultSignalDetectionPhase(phase: CommonTypes.PhaseType): void;
  setSelectedSortType(sortType: AnalystWorkspaceTypes.WaveformSortType): void;
  setAnalystNumberOfWaveforms(value: number, valueAsString?: string): void;
  setMaskDisplayFilters(key: string, maskDisplayFilter: MaskDisplayFilter): void;
  setWaveformAlignment(
    alignWaveformsOn: AlignWaveformsOn,
    phaseToAlignOn: CommonTypes.PhaseType,
    showPredictedPhases: boolean
  ): void;
  toggleMeasureWindow(): void;
  setShowPredictedPhases(showPredicted: boolean): void;
  pan(panDirection: PanType): void;
  onKeyPress(
    e: React.KeyboardEvent<HTMLDivElement>,
    clientX?: number,
    clientY?: number,
    channelId?: string,
    timeSecs?: number
  ): void;
  amplitudeScaleOption: AmplitudeScalingOptions;
  fixedScaleVal: FixedScaleValue;
  setAmplitudeScaleOption: (option: AmplitudeScalingOptions) => void;
  setFixedScaleVal: (val: FixedScaleValue) => void;
}

export interface WaveformDisplayControlsState {
  hasMounted: boolean;
}
