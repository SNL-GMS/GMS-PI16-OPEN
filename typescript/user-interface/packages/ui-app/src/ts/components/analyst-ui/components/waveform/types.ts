import { CommonTypes, EventTypes, WaveformTypes, WorkflowTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import { AnalystWorkspaceTypes, CommonWorkspaceTypes } from '@gms/ui-state';
import { WeavessTypes } from '@gms/weavess-core';
import Immutable from 'immutable';
import { MutationFunction } from 'react-query';

import { QcMaskDisplayFilters } from '~components/analyst-ui/config';
import { QueryAndMutationTypes } from '~components/client-interface';

import {
  AmplitudeScalingOptions,
  FixedScaleValue
} from './components/waveform-controls/scaling-options';

export enum PhaseAlignments {
  PREDICTED_PHASE = 'Predicted',
  OBSERVED_PHASE = 'Observed'
}

export enum AlignWaveformsOn {
  TIME = 'Time',
  PREDICTED_PHASE = 'Predicted',
  OBSERVED_PHASE = 'Observed'
}

export enum KeyDirection {
  UP = 'Up',
  DOWN = 'Down',
  LEFT = 'Left',
  RIGHT = 'Right'
}

/**
 * Waveform Display display state.
 * keep track of selected channels & signal detections
 */
export interface WaveformDisplayState {
  weavessStations: WaveformWeavessStation[];
  currentTimeInterval: CommonTypes.TimeRange;
  // because the user may load more waveform
  // data than the currently opened time interval
  viewableInterval: WeavessTypes.TimeRange;
  loadingWaveforms: boolean;
  loadingWaveformsPercentComplete: number;
  maskDisplayFilters: QcMaskDisplayFilters;
  analystNumberOfWaveforms: number;
  currentOpenEventId: string;
  showPredictedPhases: boolean;
  alignWaveformsOn: AlignWaveformsOn;
  phaseToAlignOn: CommonTypes.PhaseType | undefined;
  isMeasureWindowVisible: boolean;
  amplitudeScaleOption: AmplitudeScalingOptions;
  fixedScaleVal: FixedScaleValue;
}

/**
 * Props mapped in from Redux state
 */
export interface WaveformDisplayReduxProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  defaultSignalDetectionPhase: CommonTypes.PhaseType;
  currentOpenEventId: string;
  selectedSdIds: string[];
  stationsVisibility: Immutable.Map<string, CommonWorkspaceTypes.StationVisibilityObject>;
  selectedStationIds: string[];
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  analysisMode: WorkflowTypes.AnalysisMode;
  measurementMode: AnalystWorkspaceTypes.MeasurementMode;
  sdIdsToShowFk: string[];
  location: AnalystWorkspaceTypes.LocationSolutionState;
  channelFilters: Immutable.Map<string, WaveformTypes.WaveformFilter>;
  openEventId: string;
  keyPressActionQueue: Immutable.Map<string, number>;
  waveformClient: AnalystWorkspaceTypes.WaveformClient;

  // callbacks
  setDefaultSignalDetectionPhase(phase: CommonTypes.PhaseType): void;
  setMode(mode: AnalystWorkspaceTypes.WaveformDisplayMode): void;
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedSdIds(idx: string[]): void;
  setSdIdsToShowFk(signalDetections: string[]): void;
  setSelectedSortType(selectedSortType: AnalystWorkspaceTypes.WaveformSortType): void;
  setChannelFilters(filters: Immutable.Map<string, WaveformTypes.WaveformFilter>);
  setMeasurementModeEntries(entries: Immutable.Map<string, boolean>): void;
  setKeyPressActionQueue(actions: Immutable.Map<string, number>): void;
  setStationsVisibility(
    stationsVisibility: Immutable.Map<string, CommonWorkspaceTypes.StationVisibilityObject>
  );
}
/**
 * Mutations used by the Waveform display
 */
export interface WaveformDisplayMutations {
  markAmplitudeMeasurementReviewed: MutationFunction<unknown, unknown>;
}

/**
 * Consolidated props type for waveform display.
 */
export type WaveformDisplayProps = WaveformDisplayReduxProps &
  WaveformDisplayMutations &
  QueryAndMutationTypes.ProcessingAnalystConfigurationQueryProps &
  QueryAndMutationTypes.EventQueryProps &
  QueryAndMutationTypes.SignalDetectionsQueryProps &
  QueryAndMutationTypes.StationDefinitionQueryProps &
  QueryAndMutationTypes.QCMaskQueryProps;
// Enum to clarify pan button interactions
export enum PanType {
  Left,
  Right
}

/**
 * Station configuration using the Waveform definition of WeavessChannel
 */
export interface WaveformWeavessStation extends WeavessTypes.Station {
  /** Default channel information for station */
  defaultChannel: WaveformWeavessChannel;

  /** Non-default channels for station */
  nonDefaultChannels?: WaveformWeavessChannel[];
}

/**
 * Channel configuration adding the isChannelVisible attribute to WeavessTypes.Channel
 */
export interface WaveformWeavessChannel extends WeavessTypes.Channel {
  isChannelVisibleInWeavess: boolean;
}
