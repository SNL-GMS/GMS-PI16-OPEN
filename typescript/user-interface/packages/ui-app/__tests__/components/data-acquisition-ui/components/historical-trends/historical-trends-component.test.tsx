import { SohTypes } from '@gms/common-model';
import { ValueType } from '@gms/common-util/lib/types/value-type';
import React from 'react';

import {
  buildHistoricalTrendsHistoryComponent,
  HistoricalTrendsHistoryComponentProps
} from '../../../../../src/ts/components/data-acquisition-ui/components/historical-trends';
import { testStationSoh } from '../../../../__data__/data-acquisition-ui/soh-overview-data';
import { sohConfiguration } from '../../../../__data__/data-acquisition-ui/soh-params-data';
import { reactQueryResult } from '../../../../__data__/test-util';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

const MOCK_TIME = 1606818240000;
global.Date.now = jest.fn(() => MOCK_TIME);
const sohConfigurationQuery = reactQueryResult;
sohConfigurationQuery.data = sohConfiguration;
const defaultHistoricalTrendsComponentProps: HistoricalTrendsHistoryComponentProps = {
  selectedStationIds: [],
  setSelectedStationIds: jest.fn(),
  sohStatus: {
    lastUpdated: 1,
    loading: false,
    error: undefined,
    isStale: false,
    stationAndStationGroupSoh: {
      isUpdateResponse: false,
      stationGroups: [],
      stationSoh: [testStationSoh]
    }
  },
  type: SohTypes.SohMonitorType.LAG,
  sohConfigurationQuery
};

describe('HistoricalTrendsComponent', () => {
  it('is buildHistoricalTrendsHistoryComponent exported', () => {
    expect(buildHistoricalTrendsHistoryComponent).toBeDefined();
  });

  it('is buildHistoricalTrendsHistoryComponent loading', () => {
    const Component = buildHistoricalTrendsHistoryComponent(
      SohTypes.SohMonitorType.LAG,
      ValueType.FLOAT,
      `title`
    );
    const mockComponent = Enzyme.shallow(
      <Component
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...defaultHistoricalTrendsComponentProps}
        sohStatus={{
          ...defaultHistoricalTrendsComponentProps.sohStatus,
          loading: true
        }}
      />
    );
    expect(mockComponent).toMatchSnapshot();
  });

  it('is buildHistoricalTrendsHistoryComponent no data', () => {
    const Component = buildHistoricalTrendsHistoryComponent(
      SohTypes.SohMonitorType.LAG,
      ValueType.FLOAT,
      `title`
    );
    const mockComponent = Enzyme.shallow(
      <Component
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...defaultHistoricalTrendsComponentProps}
        sohStatus={{
          ...defaultHistoricalTrendsComponentProps.sohStatus,
          stationAndStationGroupSoh: undefined
        }}
      />
    );
    expect(mockComponent).toMatchSnapshot();
  });

  it('is buildHistoricalTrendsHistoryComponent rendered', () => {
    const Component = buildHistoricalTrendsHistoryComponent(
      SohTypes.SohMonitorType.LAG,
      ValueType.FLOAT,
      `title`
    );
    // eslint-disable-next-line react/jsx-props-no-spreading
    const mockComponent = Enzyme.shallow(<Component {...defaultHistoricalTrendsComponentProps} />);
    expect(mockComponent).toMatchSnapshot();
  });
});
