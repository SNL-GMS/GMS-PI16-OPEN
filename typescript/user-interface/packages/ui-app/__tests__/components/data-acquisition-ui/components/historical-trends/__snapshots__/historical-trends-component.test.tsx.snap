// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoricalTrendsComponent is buildHistoricalTrendsHistoryComponent loading 1`] = `
<BaseDisplay
  className="history-display top-level-container scroll-box scroll-box--y"
>
  <HistoricalTrendsHistoryPanel
    displaySubtitle="title"
    monitorType="LAG"
    sohHistoricalDurations={
      Array [
        21600000,
        43200000,
        86400000,
      ]
    }
    sohStatus={
      Object {
        "error": undefined,
        "isStale": false,
        "lastUpdated": 1,
        "loading": true,
        "stationAndStationGroupSoh": Object {
          "isUpdateResponse": false,
          "stationGroups": Array [],
          "stationSoh": Array [
            Object {
              "allStationAggregates": Array [],
              "channelSohs": Array [
                Object {
                  "allSohMonitorValueAndStatuses": Array [
                    Object {
                      "contributing": false,
                      "hasUnacknowledgedChanges": false,
                      "monitorType": "LAG",
                      "quietUntilMs": 1,
                      "status": "GOOD",
                      "thresholdBad": 10,
                      "thresholdMarginal": 1,
                      "value": 1,
                      "valuePresent": true,
                    },
                  ],
                  "channelName": "AAA111",
                  "channelSohStatus": "GOOD",
                },
              ],
              "id": "1",
              "needsAcknowledgement": true,
              "needsAttention": true,
              "sohStatusSummary": "MARGINAL",
              "stationGroups": Array [
                Object {
                  "groupName": "GroupA",
                  "sohStationCapability": "BAD",
                  "stationName": "Test",
                },
              ],
              "stationName": "Test",
              "statusContributors": Array [],
              "time": 123456789,
              "uuid": "2",
            },
          ],
        },
      }
    }
    valueType="FLOAT"
  />
</BaseDisplay>
`;

exports[`HistoricalTrendsComponent is buildHistoricalTrendsHistoryComponent no data 1`] = `
<BaseDisplay
  className="history-display top-level-container scroll-box scroll-box--y"
>
  <HistoricalTrendsHistoryPanel
    displaySubtitle="title"
    monitorType="LAG"
    sohHistoricalDurations={
      Array [
        21600000,
        43200000,
        86400000,
      ]
    }
    sohStatus={
      Object {
        "error": undefined,
        "isStale": false,
        "lastUpdated": 1,
        "loading": false,
        "stationAndStationGroupSoh": undefined,
      }
    }
    valueType="FLOAT"
  />
</BaseDisplay>
`;

exports[`HistoricalTrendsComponent is buildHistoricalTrendsHistoryComponent rendered 1`] = `
<BaseDisplay
  className="history-display top-level-container scroll-box scroll-box--y"
>
  <HistoricalTrendsHistoryPanel
    displaySubtitle="title"
    monitorType="LAG"
    sohHistoricalDurations={
      Array [
        21600000,
        43200000,
        86400000,
      ]
    }
    sohStatus={
      Object {
        "error": undefined,
        "isStale": false,
        "lastUpdated": 1,
        "loading": false,
        "stationAndStationGroupSoh": Object {
          "isUpdateResponse": false,
          "stationGroups": Array [],
          "stationSoh": Array [
            Object {
              "allStationAggregates": Array [],
              "channelSohs": Array [
                Object {
                  "allSohMonitorValueAndStatuses": Array [
                    Object {
                      "contributing": false,
                      "hasUnacknowledgedChanges": false,
                      "monitorType": "LAG",
                      "quietUntilMs": 1,
                      "status": "GOOD",
                      "thresholdBad": 10,
                      "thresholdMarginal": 1,
                      "value": 1,
                      "valuePresent": true,
                    },
                  ],
                  "channelName": "AAA111",
                  "channelSohStatus": "GOOD",
                },
              ],
              "id": "1",
              "needsAcknowledgement": true,
              "needsAttention": true,
              "sohStatusSummary": "MARGINAL",
              "stationGroups": Array [
                Object {
                  "groupName": "GroupA",
                  "sohStationCapability": "BAD",
                  "stationName": "Test",
                },
              ],
              "stationName": "Test",
              "statusContributors": Array [],
              "time": 123456789,
              "uuid": "2",
            },
          ],
        },
      }
    }
    valueType="FLOAT"
  />
</BaseDisplay>
`;
